shippo/tests/http_test.rs

```diff
@@ -0,0 +1,261 @@
+#![recursion_limit = "256"]
+
+use httpmock::MockServer;
+use reqwest::Url;
+
+use shippo::Shippo;
+
+mod data;
+
+#[tokio::test]
+async fn list_shipments() {
+    let server = MockServer::start();
+    let client = Shippo::new("token123");
+    let mock_url = Url::parse(&server.base_url()).unwrap();
+    let mock = server.mock(|when, then| {
+        when.method("GET")
+            .path("/shipments")
+            .header("authorization", String::from("ShippoToken token123")) // httpmock masks auth keys
+            .header("Content-Type", "application/json");
+        then.status(200).json_body(data::list_shipments().json_body);
+    });
+
+    // Act
+    let result = client.base_url(mock_url).list_shipments().await;
+
+    // Assert
+    mock.assert();
+    assert!(result.is_ok());
+    let response = result.unwrap();
+    assert_eq!(response.len(), 1);
+    assert_eq!(response[0], data::list_shipments().deserialized[0])
+}
+
+#[tokio::test]
+async fn create_shipment() {
+    // Arrange
+    let server = MockServer::start();
+    let mock_url = Url::parse(&server.base_url()).unwrap();
```

Comment:
Going to check if I can just initialize this data once to keep the tests more DRY

Reply:
no way to make this more dry.